src/st/no0/e_elevator.c

```diff
@@ -261,4 +261,113 @@ void func_us_801C26B8(Entity* self) {
     }
 }
 
-INCLUDE_ASM("st/no0/nonmatchings/e_elevator", func_us_801C27A4);
+void func_us_801C27A4(Entity* self) {
+    Primitive* prim;
+    s32 primIndex;
+    s16 yOffset;
+    Entity* player = &PLAYER;
+#ifndef VERSION_PSP
+    s16 step;
+#endif
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitElevator);
+        self->animCurFrame = 3;
+        self->zPriority = player->zPriority + 2;
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 12);
+        if (primIndex != -1) {
+            self->flags |= FLAG_HAS_PRIMS;
+            self->primIndex = primIndex;
+            prim = &g_PrimBuf[primIndex];
+            self->ext.prim = prim;
+            while (prim != NULL) {
+                prim->tpage = 0x12;
+                prim->u0 = prim->u2 = 0x50;
+                prim->u1 = prim->u3 = 0x60;
+                prim->v0 = prim->v1 = 6;
+                prim->v2 = prim->v3 = 0x26;
+                prim->clut = 0x223;
+                prim->priority = 0x6A;
+                prim->drawMode = DRAW_HIDE;
+                prim = prim->next;
+            }
+        } else {
+            DestroyEntity(self);
+            return;
+        }
+
+#if defined(VERSION_PSP)
```

Comment:
Not sure if I'm missing something but the psp version matches on ps1 for me?

Reply:
Right you are, I refactored this code so many times I must have not tried this variation for some reason. Updated